include/boost/gil/image_processing/numeric.hpp

```diff
@@ -295,6 +295,120 @@ inline void compute_hessian_entries(
     detail::convolve_2d(dy, sobel_y, ddyy);
 }
 
+namespace detail {
+enum class direction: std::size_t {
+    north = 0,
+    south = 1,
+    west = 2,
+    east = 3,
+    north_east = 4,
+    south_east = 5,
+    south_west = 6,
+    north_west = 7
+};
+
+template <typename OutputView, typename InputView>
+void compute_nabla(InputView view, const std::vector<OutputView>& nabla) {
+    constexpr std::ptrdiff_t input_num_channels = num_channels<InputView>{};
+    static_assert(num_channels<OutputView>{} == input_num_channels, "input and output views must have the same amount of channels");
+    for (std::ptrdiff_t y = 1; y < view.height() - 1; ++y)
+    {
+        for (std::ptrdiff_t x = 1; x < view.width() - 1; ++x)
+        {
+            for (std::ptrdiff_t channel_index = 0; channel_index < input_num_channels; ++channel_index)
+            {
+                nabla[(std::size_t)direction::north](x, y)[channel_index] = view(x, y - 1)[channel_index] - view(x, y)[channel_index];
```

Comment:
I'd rather avoid `(std::size_t)` cast here.

Reply:
It will not compile without a cast, as it is `enum class` type. I thought about omitting `class` keyword, but I'm afraid I will pay for it layer.

Reply:
Since these are used as numbers, as indexes, and not as enumerators with e.g. `switch` statement, I'd use plain `enum`  inside structure as scope or constants within dedicated namespace:

```cpp
struct direction
{
    enum { north = 0, south = 1, ... }
};
```

or

```cpp
namespace direction {
    constexpr unsigned int north = 0;
    constexpr unsigned int south = 1;
}
```

Reply:
I put it inside a namespace and made it nameless enum. Feeling bad about it, but it works.